### PR TITLE
refactor: updated types for `ChargebeeInstance`

### PR DESCRIPTION
**Pull request by cb-farhanh**
#### Description
- Updated types for `ChargebeeInstance`.

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -23,12 +24,321 @@ type InitOptions = {
...
+  decrementPlanQuantity(): Product;
+  addAddon(addon: AddOn | string): Product;
+  removeAddon(addon: AddOn | string): Product;
+  setAddons(addons: Array<AddOn>): Product;
+  incrementAddonQty(addonId: string): Product;
+  decrementAddonQty(addonId: string): Product;
+  addCoupon(couponCode: string): Product;
+  removeCoupon(couponCode: string): Product;
+  setCustomData(
+    data: object /** Object containing subs custom fields */
```
Comment (cb-dinesh) [inline index.d.ts:100]: Object

File: index.d.ts [hunk 2]
```diff
@@ -23,12 +24,321 @@ type InitOptions = {
...
+  layout: Layout;
+  loaded: Function;
+  error: Function;
+  success(hostedPageId: string): void;
+  close: Function;
+  step(currentStep: string): void;
+};
+
+interface AddressDetails {
+  first_name: string;
```
Comment (cb-dinesh) [inline index.d.ts]: Mark as optional?

File: index.d.ts [hunk 3]
```diff
@@ -23,12 +24,321 @@ type InitOptions = {
...
+interface Cart {
+  replaceProduct(product: Product): Cart;
+  proceedToCheckout(): void;
+  setBusinessEntity(businessEntityId: string): Promise<void>;
+  setAffiliateToken(token: string): Cart;
+  setCustomer(customer: Customer): Cart;
+  setShippingAddress(shippingAddress: ShippingAddress): Cart;
+}
+
+interface PortalOpenOptions {
```
Comment (cb-dinesh) [inline index.d.ts]: Should this directly be named PortalCallbacks

File: index.d.ts [hunk 4]
```diff
@@ -23,12 +24,321 @@ type InitOptions = {
...
+  firstName?: string;
+  lastName?: string;
+};
+
+type PaymentInfo = {
+  element?: any;
+  card?: CardInfo;
+  tokenizer?: Function;
+  cbToken?: string;
+  cardComponent?: string;
```
Comment (cb-dinesh) [inline index.d.ts:190]: boolean or string?

File: index.d.ts (hunk 2 above)
Comment (cb-farhanh) [reply index.d.ts]: Done.

File: index.d.ts (hunk 3 above)
Comment (cb-farhanh) [reply index.d.ts]: Done.

**Review by cb-dinesh — APPROVED** [review]

**Comment by jaska120** [comment]
How could this PR get merged and new version published? It would be great to rely on official Chargebee typings instead of own ones.

**Pull request merged**